src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryOutputFormat.java

```diff
@@ -392,26 +381,33 @@ private void importFromGcs(String projectId, TableReference tableRef, @Nullable
         LOG.info("Using schema '{}' for the load job config.", loadConfig.getSchema());
       }
 
-      JobConfiguration config = new JobConfiguration();
-      config.setLoad(loadConfig);
 
       // Get the dataset to determine the location
       Dataset dataset =
         bigQueryHelper.getRawBigquery().datasets().get(tableRef.getProjectId(), tableRef.getDatasetId()).execute();
 
-      JobReference jobReference =
-        new JobReference().setProjectId(projectId)
-          .setJobId(jobId)
-          .setLocation(dataset.getLocation());
-      Job job = new Job();
-      job.setConfiguration(config);
-      job.setJobReference(jobReference);
+      String temporaryTableName = tableRef.getTableId() + "_"
+        + UUID.randomUUID().toString().replaceAll("-", "_");
+      temporaryTableReference = new TableReference()
+        .setDatasetId(tableRef.getDatasetId())
+        .setProjectId(tableRef.getProjectId())
+        .setTableId(temporaryTableName);
 
-      // Insert and run job.
-      bigQueryHelper.insertJobOrFetchDuplicate(projectId, job);
+      loadConfig.setDestinationTable(temporaryTableReference);
+      loadConfig.setWriteDisposition(JobInfo.WriteDisposition.WRITE_APPEND.toString()); // Always append to temp table
 
-      // Poll until job is complete.
-      waitForJobCompletion(bigQueryHelper.getRawBigquery(), projectId, jobReference);
+      List<List<String>> gcsPathsInBatches = Lists.partition(gcsPaths, BQ_IMPORT_MAX_BATCH_SIZE);
+
+      int jobcount = 1;
```

Comment:
add a comment here explaining what this logic does (load in batches of up to 10k records)

Reply:
done